test/spec/RunSpec.hs

```diff
@@ -9,265 +9,243 @@ import Data.String.Interpolate.Util (unindent)
 import System.Directory
 import System.Exit (ExitCode (..))
 import Test.Hspec
-import Test.Mockery.Directory
-import Test.Mockery.Environment (withModifiedEnvironment)
 import TestUtils
 
 spec :: Spec
 spec =
-  describe "run" $ around onTestFailureLogger $ do
+  describe "run" $ withRunGarn $ do
     repoDir <- runIO getCurrentDirectory
-    around_
-      ( withModifiedEnvironment [("NIX_CONFIG", "experimental-features =")]
-          . inTempDirectory
-      )
-      $ do
-        it "runs a simple Haskell program" $ \onTestFailureLog -> do
-          writeHaskellProject repoDir
-          output <- runGarn ["run", "foo.garn-test"] "" repoDir Nothing
-          onTestFailureLog output
-          stdout output `shouldBe` "haskell test output\n"
-        it "writes flake.{lock,nix}, but no other files" $ \onTestFailureLog -> do
-          writeHaskellProject repoDir
-          filesBefore <- listDirectory "."
-          output <- runGarn ["run", "foo.garn-test"] "" repoDir Nothing
-          onTestFailureLog output
-          filesAfter <- sort <$> listDirectory "."
-          filesAfter `shouldBe` sort (filesBefore ++ ["flake.lock", "flake.nix"])
-        it "runs arbitrary executables" $ \onTestFailureLog -> do
-          writeFile
-            "garn.ts"
+    it "runs a simple Haskell program" $ \runGarn -> do
+      writeHaskellProject repoDir
+      output <- runGarn ["run", "foo.garn-test"]
+      stdout output `shouldBe` "haskell test output\n"
+
+    it "writes flake.{lock,nix}, but no other files" $ \runGarn -> do
+      writeHaskellProject repoDir
+      filesBefore <- listDirectory "."
+      _ <- runGarn ["run", "foo.garn-test"]
+      filesAfter <- sort <$> listDirectory "."
+      filesAfter `shouldBe` sort (filesBefore ++ ["flake.lock", "flake.nix"])
+
+    it "runs arbitrary executables" $ \runGarn -> do
+      writeFile
+        "garn.ts"
+        [i|
+          import * as garn from "#{repoDir}/ts/mod.ts"
+
+          export const main = garn.shell("echo foobarbaz");
+        |]
+      output <- runGarn ["run", "main"]
+      stdout output `shouldBe` "foobarbaz\n"
+      exitCode output `shouldBe` ExitSuccess
+
+    it "propagates the exit status of the child process" $ \runGarn -> do
```

Comment:
I'm assuming the only change here is indentation?

Reply:
Well, and removing the call to `onTestFailureLog` and calling the passed in `runGarn`. But yeah, it's a very mechanical refactoring.